server/chat-plugins/sample-teams.ts

```diff
@@ -0,0 +1,515 @@
+/**
+ * A sample teams plugin
+ *
+ * @author Kris
+ */
+
+import {FS, Utils} from "../../lib";
+
+const SAMPLE_TEAMS = 'config/chat-plugins/sample-teams.json';
+
+interface SampleTeamsFile {
+	whitelist: {[formatid: string]: RoomID[]};
+	/** Teams are stored in the packed format */
+	teams: {
+		[formatid: string]: {
+			uncategorized: {[k: string]: string},
+			[category: string]: {[teamName: string]: string},
+		},
+	};
+}
+
+const file: SampleTeamsFile = JSON.parse(FS(SAMPLE_TEAMS).readIfExistsSync() || `{"whitelist": {}, "teams": {}}`);
+
+function save() {
+	FS(SAMPLE_TEAMS).writeUpdate(() => JSON.stringify(file));
+}
+
+if (!file.whitelist) file.whitelist = {};
+if (!file.teams) file.teams = {};
+for (const formatid in file.teams) {
+	if (!file.teams[formatid].uncategorized) file.teams[formatid].uncategorized = {};
+}
+save();
+
+const SampleTeams = new class SampleTeams {
+	isRoomStaff(user: User, roomids: RoomID[]) {
```

Comment:
This seems like something that could be reused in other parts of the code? It can also be `static`

Reply:
Probably shouldn't be static? It's a singleton instance, I don't think that cooperates.

Reply:
It's mostly just a helper function, the `checkPermissions` function does everything necessary